### DISCUSSION TITLE
rule-based matching named entitys

### DISCUSSION THREAD

**Discussion by the1gofer** [explosion/spaCy#6915] [Help: Coding & Implementations] [answered]
I can't find this mentioned in the documentation.

is it possible to include a named entity in a rule?  for example "some text born on june 1, 2020".  assuming I don't know the exact format of the date.   Could I do something like:
`
pattern = [{"LOWER": "was born"}, {"label": "DATE"}]`

**Comment by svlandeg** [comment, marked as answer]
Yes, you can use `ENT_TYPE`. You can find a list of available attributes here: https://spacy.io/usage/rule-based-matching#adding-patterns-attributes

As an example code snippet:

```
    import spacy
    from spacy.matcher import Matcher

    nlp = spacy.load("en_core_web_sm")
    matcher = Matcher(nlp.vocab)
    pattern = [{"LOWER": "was"}, {"LOWER": "born"}, {"LOWER": "on"}, {"ENT_TYPE": "DATE"}]
    matcher.add("BORN", [pattern])
    text = "Some text was born on june 1, 2020"
    doc = nlp(text)
    matches = matcher(doc)
```
This will give you a match for token 2 'till 6.

Note that you need to separate each token in the pattern (and not keep "was born" together).

**Reply by the1gofer** [reply]
Thanks.  I saw that link.  Where do I find a list of the labels?

**Reply by svlandeg** [reply]
Those depend on the model. You could for instance check https://spacy.io/models/en, and expand the "Label Scheme" section for the model you're using, then check the NER labels :-)

**Reply by raqibhayder** [reply]
Can we also use custom entity types? For example, the ones we train, or add through entity ruler?

**Reply by svlandeg** [reply]
Yes, you definitely can!

Perhaps I should have been more clear in my earlier post. You can basically use all entity labels that are being predicted by your pipeline. If you are downloading a pretrained pipeline like `en_core_web_sm`, typically you'll be using the labels from the pretrained NER component, and you can find those in the documentation.

For any NER component (pretrained or your own), you can also call `nlp.get_pipe("ner").labels` to get an overview of the available NER labels.

And you can use entity labels from any other component that create entities. The `entity_ruler` also has a `labels` property if you want to check which ones are available. Or alternatively, you could run through all the `doc.ents` and check which labels were predicted by your custom pipeline ;-)
